### PR TITLE
FirstOrder policy added

### PR DESCRIPTION
**Pull request by hcorg**

### CODE REVIEW

**Comment by hcorg** [comment]
there is one semi-ugly TODO left for discussion (when adding numbers to Domain.Quality, for ensuring proper ordering of qualities, I'm forced to add the same numbers to Dto.Quality, which is maybe not that ugly, but suggests implementation detail to experiment subjects - small, but still - I want second opinion :) )

All Hail Snoke!

File: Signals/Domain/MissingValuePolicy/FirstOrderMissingValuePolicy.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+            return (Quality)Math.Max((int)a, (int)b);
+        }
+
+        private Datum<T> Interpolate(Datum<T> older, Datum<T> newer, DateTime currentTs)
+        {
+            var q = GetMinQuality(older.Quality, newer.Quality);
+            if (q == Quality.None)
+                return Datum<T>.CreateNone(Signal, currentTs);
+            var timeSpan = (newer.Timestamp - older.Timestamp).TotalSeconds;
+            var valueSpan = (dynamic)newer.Value - (dynamic)older.Value; // yuck!
```
Comment (bartoszkp) [inline Signals/Domain/MissingValuePolicy/FirstOrderMissingValuePolicy.cs]: Problem for `string`. I guess it needs some discussion: zero-order may have sense for strings, first order definitely not - so are we assuming here, that it was disallowed in UI? Or perhaps our `SetMissingValuePolicy` should throw when trying to configure string signal with this policy?

Comment (hcorg) [inline Signals/Domain/MissingValuePolicy/FirstOrderMissingValuePolicy.cs]: yeah....that's a second TODO I have for second discussion ;) (3rd TODO later - int rounding...)

IMHO - `Set` should disallow it

**Comment by hcorg** [comment]
TODO resolved in separate branch

do we care at this moment about int rounding?

**Comment by bartoszkp** [comment]
Adding some specific, required int rounding semantics is a good idea to provide some kind of challenge to test subjects. We can consider doing in it as a next step.

**Pull request merged**